### ISSUE TITLE
Segmentation fault when running 

### ISSUE THREAD

**Issue by nimashiri** [tensorflow/tensorflow#59135]
### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.11.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
Segmentation fault
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import nn_ops
try:
  arg_0_tensor = tf.random.uniform([3, 30, 50, 3], dtype=tf.float64)
  arg_0 = tf.identity(arg_0_tensor)
  arg_1_0 = 2
  arg_1_1 = 3
  arg_1_2 = 2
  arg_1_3 = 1
  arg_1 = [arg_1_0,arg_1_1,arg_1_2,arg_1_3,]
  arg_2 = True
  arg_3 = True
  seed = 341261001
  out = nn_ops.fractional_avg_pool_v2(arg_0,arg_1,arg_2,arg_3,seed=seed,)
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: 0 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: 0 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -978 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: 0 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: 0 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -39.2679482 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: 0 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -7 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -1 [Op:FractionalAvgPool]
Error:Value for attr 'T' of bool is not in the list of allowed values: float, double, int32, int64
	; NodeDef: {{node FractionalAvgPool}}; Op<name=FractionalAvgPool; signature=value:T -> output:T, row_pooling_sequence:int64, col_pooling_sequence:int64; attr=pooling_ratio:list(float),min=4; attr=pseudo_random:bool,default=false; attr=overlapping:bool,default=false; attr=deterministic:bool,default=false; attr=seed:int,default=0; attr=seed2:int,default=0; attr=T:type,allowed=[DT_FLOAT, DT_DOUBLE, DT_INT32, DT_INT64]> [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -1 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -23.2679501 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -1 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -21.5857868 [Op:FractionalAvgPool]
Error:Expected bool for argument 'overlapping' not 'zeros'.
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: 0 [Op:FractionalAvgPool]
Error:Value for attr 'T' of complex128 is not in the list of allowed values: float, double, int32, int64
	; NodeDef: {{node FractionalAvgPool}}; Op<name=FractionalAvgPool; signature=value:T -> output:T, row_pooling_sequence:int64, col_pooling_sequence:int64; attr=pooling_ratio:list(float),min=4; attr=pseudo_random:bool,default=false; attr=overlapping:bool,default=false; attr=deterministic:bool,default=false; attr=seed:int,default=0; attr=seed2:int,default=0; attr=T:type,allowed=[DT_FLOAT, DT_DOUBLE, DT_INT32, DT_INT64]> [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -26.5857868 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -58.5857849 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: 0 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -63 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -1 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: 0 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: 0 [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} pooling_ratio cannot be smaller than 1, got: -1 [Op:FractionalAvgPool]
Error:Expected bool for argument 'overlapping' not ''.
Error:Expected float for argument 'pooling_ratio' not ''.
Segmentation fault

```
```

**Comment by mohantym** [comment]
Hi @nimashiri !
Thanks for sharing your observation w.r.to  nn_ops


@SuryanarayanaY !
Could you look at this bug.
Sessions are  crashing for both [2.10 ](https://colab.sandbox.google.com/gist/mohantym/55ccb2a0d6419b8a468e70cc17a96578/git_59135.ipynb#scrollTo=OymomtMQGTBs)and [2.11](https://colab.sandbox.google.com/gist/mohantym/55ccb2a0d6419b8a468e70cc17a96578/git_59135.ipynb#scrollTo=OymomtMQGTBs) and [nightly ](https://colab.sandbox.google.com/gist/mohantym/d18c309b639a0ab709706f121359f9cf/git_59135.ipynb#scrollTo=OymomtMQGTBs)in Colab under GPU environment but giving results under CPU environment.




Thank you!

**Comment by SuryanarayanaY** [comment]
Tested the code on Ubuntu VM and result is `Aborted (core dumped)` as attached below.
<img width="1487" alt="Screenshot 2023-01-11 at 7 37 27 PM" src="https://user-images.githubusercontent.com/116063290/211827367-382a6eae-c9dd-4ec6-9a4a-9592b67f0a46.png">
Tested with similar high level API tf.nn.fractional_avg_pool and observed same behaviour.
<img width="1502" alt="Screenshot 2023-01-11 at 7 43 19 PM" src="https://user-images.githubusercontent.com/116063290/211829279-b8a9a516-9a73-48a1-9e4b-d27e780be1d1.png">

With tf-nightly also observed same behaviour.

<img width="1504" alt="Screenshot 2023-01-11 at 7 45 11 PM" src="https://user-images.githubusercontent.com/116063290/211829428-db31a4d7-77cb-4f64-a6b3-5cbe3738e676.png">

**Comment by nimashiri** [comment]
Also this one:

```
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import nn_ops
try:
  arg_0_tensor = tf.random.uniform([1, 6, 6, 1], minval=-256, maxval=257, dtype=tf.int64)
  arg_0 = tf.identity(arg_0_tensor)
  arg_1_0 = -1
  arg_1_1 = -30.26794919243112
  arg_1_2 = -21.585786437626904
  arg_1_3 = 1
  arg_1 = [arg_1_0,arg_1_1,arg_1_2,arg_1_3,]
  arg_2 = False
  arg_3 = True
  seed = 341261000
  out = nn_ops.fractional_avg_pool_v2(arg_0,arg_1,arg_2,arg_3,seed=seed,)
except Exception as e:
  print("Error:"+str(e))
```

**Comment by zndr27** [comment]
I'm having the same issue

**Comment by mihaimaruseac** [comment]
@learning-to-play

**Comment by sachinprasadhs** [comment]
Hi, In Tensorflow 2.12 it is not showing any segmentation fault, instead it's throwing an error as 
`Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} Fractional average pooling is not yet supported on the batch nor channel dimension. [Op:FractionalAvgPool]` please find the attached Gist [here](https://gist.github.com/sachinprasadhs/4a592bdce9051cc2e6a90db5f93331d4) for reference. Thanks!

**Comment by dmc1778** [comment]
> Hi, In Tensorflow 2.12 it is not showing any segmentation fault, instead it's throwing an error as `Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} Fractional average pooling is not yet supported on the batch nor channel dimension. [Op:FractionalAvgPool]` please find the attached Gist [here](https://gist.github.com/sachinprasadhs/4a592bdce9051cc2e6a90db5f93331d4) for reference. Thanks!

Hi, this bug is shown in 2.11.0 not 2.12.

**Comment by mihaimaruseac** [comment]
2.12 has it fixed, so this can be closed
